Upgrade: Tracking WP.com connection status

sergeymitr opened the pull request:
#### Changes proposed in this Pull Request:
The PR adds two properties for the existing event `jetpack_wpa_click`:

- `is_user_wpcom_connected`
- `is_connection_owner`

Clicks on the "Upgrade" banners in Jetpack Dashboard don't properly track if the user is connected to WP.com
The existing properties (`_ui`, `_ut`) get potentially overwritten by the ETL process, so we can't rely on them for this purpose.

The goal is to find out how many non-connected secondary users try to upgrade and end up in the broken connection flow.

#### Jetpack product discussion
p9dueE-1M4-p2

#### Does this pull request change what data or activity we track or use?
The PR slightly modifies the data we track in the Jetpack Dashboard by adding two new properties to the existing events:
- Is the user connected to WP.com?
- Is this the connection owner?

No new event tracking added.

#### Testing instructions:
1. Connect your site using Jetpack Free plan.
2. Open the "Network" browser dev tool and filter by `jetpack_wpa_click`.
3. Go to the Dashboard and click on the "Scan -> Upgrade".
4. Check the network tab and confirm that the event `jetpack_wpa_click` has been sent with two properties: `is_user_wpcom_connected: yes` and `is_connection_owner: yes`.
5. Log into your site as another admin user, not connected to WP.com
6. Go to Jetpack Dashboard, filter network tab by `jetpack_wpa_click`.
7. Click on "Get a faster resolution to your support questions -> Upgrade".
8. Check the network tab and confirm that the event `jetpack_wpa_click` has been sent with two properties: `is_user_wpcom_connected: no` and `is_connection_owner: no`.

You can use the "Tracks Live" and filter by the event `jetpack_wpa_click` to find your tracks and confirm that the data was recorded correctly.

#### Proposed changelog entry for your changes:
n/a

sergeymitr committed:
Upgrade: Tracking if user is connected.

Clicks on the "Upgrade" banners don't properly track if the user is connected to WP.com
The existing properties (`_ui`, `_ut`) get overwritten by the ETL process.

The goal is to find out how many non-connected secondary users try to upgrade and end up in the broken connection flow.

brbrr commented:
Do you think it's worth checking that new new data is properly received/stored by the server, to make sure that the whole flow is functional?

sergeymitr commented:
@brbrr I tried to keep the testing instructions as simple as possible, for easier review 🙂 
I checked that it's getting saved correctly, so that's probably excessive.

But you're right, I'll add that info in the instructions.

leogermani commented:
@sergeymitr what exactly `wpcom_connected` means in this context? Is it that the user is logged in? Or is it the site being connected (having a blog token)?

Is it something we can try to choose a less ambiguous name for?

sergeymitr committed:
Renamed the `wpcom_connected` property for better clarity.

Property name `wpcom_connected` is ambiguous, renamed to `is_user_wpcom_connected`.

sergeymitr commented:
@leogermani good catch, thanks for pointing this out.
Fixed.

kbrown9 reviewed (approved):
This works well for me! I tested using the provided testing instructions, and the new properties were set as expected. Also, the tracks events were generated as expected.

jeherve commented:
Cherry-picked to `branch-9.0` in 21639f7364c1808479b6ac9045349f33fbbe4092
